### PR TITLE
[Draft] Add move option 

### PR DESCRIPTION
**Pull request by saba-phoenix**
This pull request is part of the work where we add a move option for the components. The movement will be relative to the position of the selected component's siblings. We can either -

- Move it one step up, or
- Move it one step down

### CODE REVIEW

File: lib/editor/Controller/index.tsx [hunk 1]
```diff
@@ -27,6 +27,7 @@ const StyledPlus = styled(Plus)`
 `;
 
 type PositionType = "before" | "after" | "inside";
+type DirectionType = "up" | "down";
```
Comment (Corei13) [inline lib/editor/Controller/index.tsx:30]: I think this can be removed since this is not being used

File: lib/editor/atoms/tree.ts [hunk 2]
```diff
@@ -417,7 +417,7 @@ export const deleteNodeAtom = atom(
         }
 
         set(selectedAtom, { component, id: parentId });
-
+        // const index = tree.nodes[parentId].children.findIndex(item => item === 'John');
```
Comment (Corei13) [inline lib/editor/atoms/tree.ts:420]: remove unnecessary comments and console.logs when cleaning up a PR

**Review by Corei13 — APPROVED** [review]

File: lib/editor/atoms/tree.ts (hunk 2 above)
Comment (saba-phoenix) [reply lib/editor/atoms/tree.ts:420]: I'll keep this in mind

**Review by Corei13 — APPROVED** [review]
Note to DJ: we should probably inject position of a node inside it's parent in the Node object, and refactor the moveNode(node, direction) function to moveNode(node, position) to be able to move a node to arbitrary position.

**Pull request merged**